### PR TITLE
feat(cli): add cliutil.LooksLikeJWT shared validator with length floor

### PR DESCRIPTION
**Pull request by npwalker**
## Intent

Adds a generator-emitted `cliutil` helper that every printed CLI now carries: `LooksLikeJWT(string) bool` and `FindJWTInCookieJar(jar) string`, both with a 150-char total / 20-char header minimum that filters out short tracking cookies (Cloudflare `__cf_bm`, CSRF tokens, segment-style A/B IDs) which share JWT's three-base64url-segment structure.

Issue: Refs #1598 (WU-2)

## Approach

The fix is a new pair of `cliutil_jwtshape.go.tmpl` + `cliutil_jwtshape_test.go.tmpl` templates wired into the existing `singleFiles` registration in `generator.go` (both call sites — the main path and the MCP path use the same map). Behavior:

- `LooksLikeJWT(s)` returns true iff: total length ≥ 150 chars, three dot-separated segments, header segment ≥ 20 chars, each segment matches base64url charset.
- `FindJWTInCookieJar(jar)` scans `"name=value; name=value; ..."` cookie jar string and returns the first value passing `LooksLikeJWT`, or `""` if none.
- Both strip a leading `Bearer ` prefix so callers can pass either the raw token or the wire value.

The helpers are **additive** — no existing template calls them. They become useful as soon as the next PR (WU-3a) adds an `auth set-token <jwt>` subcommand and uses `LooksLikeJWT` as its validator. The hand-authored Factor75 printed CLI also benefits when it's regen-merged — `findJWTInCookies` there can be deleted in favor of `cliutil.FindJWTInCookieJar`.

## Scope

Primary area: `comp:generator`

Why this belongs in this repo: the failure shape — sniffed CLIs hand-rolling a too-loose JWT shape check and saving short tracking cookies as access tokens — recurs across any Cloudflare-fronted site. Centralizing the helper in the generator templates means every future printed CLI inherits the corrected check.

## Risk

- **Generated output (additive):** every fresh print now emits `internal/cliutil/jwtshape.go` + `jwtshape_test.go`. Existing rendered output is unchanged because no current template calls into the new helpers.
- **`TestGenerateProjectsCompile`:** the `expectedFiles` counts for `stytch` / `clerk` / `loops` fixtures bumped by 2 (from 60/65/62 to 62/67/64) — matched in this PR.
- **`mustInclude`:** added the two new paths so a future rename / move surface-breakage is caught.
- **No regen / reprint required** for existing CLIs to benefit — the helpers ship next time those CLIs regenerate. The Factor75 printed CLI continues to work today via its hand-authored `looksLikeJWT`; replacing that with `cliutil.LooksLikeJWT` is a follow-up library-side change, not part of this PR.

## Output Contract

`internal/cliutil/jwtshape.go` (new) — always emitted. Exports `LooksLikeJWT(string) bool` and `FindJWTInCookieJar(string) string`. Length floors are unexported constants (`minJWTTotalLen`, `minJWTHeaderLen`); deliberately not configurable so the shape check is consistent across every printed CLI.

`internal/cliutil/jwtshape_test.go` (new) — always emitted. Table-driven tests cover the Factor75 false-positive case (31-char `01KRPVRYA2SNQT9BAGD6984WAG_.tt.1` rejected), a real Auth0 RS256 token (synthetic but realistically-sized — accepted), and the boundary case at the 150-char floor (just-under rejected, at-floor accepted).

## Verification

- `go build ./...` — passes
- `go test ./...` — passes (full suite, ~5 min)
- `go test ./internal/generator/... -run TestGenerate` — passes (after bumping `expectedFiles` and adding new paths to `mustInclude`)
- `scripts/golden.sh verify` — 20/20 pass (no rendered-output changes since helpers are additive)
- `go fmt ./...` — clean
- `golangci-lint` — not run locally (not installed); CI will run it

## AI / Automation Disclosure

- [x] Human-reviewed: AI or automation was used, and a human reviewed the work for intent, fit, and obvious issues before submission

### CODE REVIEW

**Comment by tmchow** [comment]
@npwalker rebase off main and address code review

File: internal/generator/templates/cliutil_jwtshape_test.go.tmpl [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	t.Parallel()
+
+	// realAuth0RS256 is a realistically-sized RS256 token (header ~64 chars,
+	// payload ~430 chars, signature ~344 chars). Synthetic — not a credential.
+	realAuth0RS256 := "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6ImFvNi1PYzAyQlZHTF9GbnBDeE5JMiJ9." +
+		strings.Repeat("a", 430) + "." + strings.Repeat("b", 344)
+
+	// minimalHS256 has header ≥ 20 chars + payload + signature, total ≥ 150.
+	// Smallest realistic JWT shape we expect to accept.
+	minimalHS256 := strings.Repeat("h", 36) + "." + strings.Repeat("p", 80) + "." + strings.Repeat("s", 40)
```
[the remark this answers is not in the document]
Comment (npwalker) [reply internal/generator/templates/cliutil_jwtshape_test.go.tmpl]: > Boundary tests are far from the 150-char floor — `"length just under floor"` totals 108 chars and `"minimal HS256 at floor"` totals 158 chars, so neither actually exercises the boundary at 149/150.

Addressed: added `atFloor149` (exactly 149 chars — must reject) and `atFloor150` (exactly 150 chars — must accept) as separate fixtures with comments documenting the byte arithmetic. The existing 158-char case stays as `realistic minimum HS256` so a regression that breaks normal-sized small JWTs surfaces independently of the boundary check. An off-by-one in the comparison or a constant change to 155 now flips one of these explicit cases.

File: internal/generator/templates/cliutil_jwtshape.go.tmpl [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		}
+		eq := strings.IndexByte(raw, '=')
+		if eq < 0 {
+			continue
+		}
+		value := raw[eq+1:]
+		if LooksLikeJWT(value) {
+			return value
+		}
+	}
```
[the remark this answers is not in the document]
Comment (npwalker) [reply internal/generator/templates/cliutil_jwtshape.go.tmpl:94]: > `FindJWTInCookieJar` returns the raw cookie value verbatim, but `LooksLikeJWT` strips the scheme prefix internally before validating. If a cookie carries the Authorization wire value as its content, the function validates it successfully but returns the prefixed form. A downstream caller passing this directly to an HTTP client would produce a double-prefixed `Authorization` header.

Addressed: `FindJWTInCookieJar` now applies `strings.TrimPrefix(..., "Bearer ")` (plus a `TrimSpace` for leading whitespace inside the value) before the `LooksLikeJWT` check, so the returned value is always the bare token. Matches `LooksLikeJWT`'s own input normalization. Added a `cookie value carries Bearer prefix` test case and updated the doc comment to make the contract explicit (callers always prepend `Bearer ` exactly once).

**Comment by npwalker** [comment]
> @npwalker rebase off main and address code review

Done — rebased onto current `main` (clean, no conflicts) and addressed both greptile P2 threads in `66622f91`:

- **Boundary tests** now hit 149/150 exactly so a constant change or comparison-operator drift surfaces immediately.
- **`FindJWTInCookieJar`** now strips a leading `Bearer ` from the cookie value before returning, so callers building an Authorization header always prepend the scheme once. New test case proves it.

Both threads resolved. `go test ./...` green, `scripts/golden.sh verify` 20/20.
